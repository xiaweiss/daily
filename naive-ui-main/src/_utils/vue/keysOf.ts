export function keysOf<T> (obj: T): Array<keyof T> {
  return Object.keys(obj) as any
}
