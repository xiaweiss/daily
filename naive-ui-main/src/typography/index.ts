export { NH1, NH2, NH3, NH4, NH5, NH6 } from './src/headers'
export type { HeaderProps as H1Props } from './src/create-header'
export type { HeaderProps as H2Props } from './src/create-header'
export type { HeaderProps as H3Props } from './src/create-header'
export type { HeaderProps as H4Props } from './src/create-header'
export type { HeaderProps as H5Props } from './src/create-header'
export type { HeaderProps as H6Props } from './src/create-header'
export { default as NA } from './src/a'
export type { AProps } from './src/a'
export { default as NP } from './src/p'
export type { PProps } from './src/p'
export { default as NBlockquote } from './src/blockquote'
export type { BlockquoteProps } from './src/blockquote'
export { default as NHr } from './src/hr'
export { default as NUl } from './src/ul'
export type { UlProps } from './src/ul'
export { default as NOl } from './src/ol'
export type { OlProps } from './src/ol'
export { default as NLi } from './src/li'
export { default as NText } from './src/text'
export type { TextProps } from './src/text'
