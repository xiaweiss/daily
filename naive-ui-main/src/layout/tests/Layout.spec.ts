import { h } from 'vue'
import { mount } from '@vue/test-utils'
import {
  NLayout,
  NLayoutContent,
  NLayoutFooter,
  NLayoutHeader,
  NLayoutSider
} from '../index'
import { NH2 } from '../../typography'
import { NCard } from '../../card'

describe('n-layout', () => {
  it('should work with import on demand', () => {
    mount(NLayout)
  })

  it('should work with Basic', async () => {
    const wrapper = mount(NLayout, {
      props: {
        'has-sider': true
      },
      slots: {
        default: () => [
          h(NLayoutHeader, null, { default: () => 'test-header' }),
          h(NLayoutContent, null, { default: () => 'test-content' }),
          h(NLayoutSider, null, { default: () => 'test-sider' }),
          h(NLayoutFooter, null, { default: () => 'test-footer' })
        ]
      }
    })

    expect(
      wrapper.find('.n-layout-scroll-container').element.children.length
    ).toBe(4)
    expect(
      wrapper
        .find('.n-layout-scroll-container')
        .element.children[0].getAttribute('class')
    ).toContain('n-layout-header')
    expect(wrapper.find('.n-layout-header').text()).toBe('test-header')
    expect(
      wrapper
        .find('.n-layout-scroll-container')
        .element.children[1].getAttribute('class')
    ).toContain('n-layout-content')
    expect(wrapper.findAll('.n-layout-scroll-container')[1].text()).toBe(
      'test-content'
    )
    expect(
      wrapper
        .find('.n-layout-scroll-container')
        .element.children[2].getAttribute('class')
    ).toContain('n-layout-sider')
    expect(wrapper.find('.n-layout-sider-scroll-container').text()).toBe(
      'test-sider'
    )
    expect(
      wrapper
        .find('.n-layout-scroll-container')
        .element.children[3].getAttribute('class')
    ).toContain('n-layout-footer')
    expect(wrapper.find('.n-layout-footer').text()).toBe('test-footer')
  })

  it('should work with `content-style` prop', async () => {
    const wrapper = mount(NLayout, {
      props: {
        'has-sider': true
      },
      slots: {
        default: () =>
          h(
            NLayoutSider,
            { 'content-style': 'padding: 24px' },
            {
              default: () => [
                h(NH2, null, { default: () => 'test1' }),
                h(NH2, null, { default: () => 'test2' })
              ]
            }
          )
      }
    })
    expect(
      wrapper.find('.n-layout-sider-scroll-container').attributes('style')
    ).toContain('padding: 24px')
  })

  it('should work with `embedded` prop', async () => {
    const wrapper = mount(NLayout, {
      props: {
        embedded: true
      },
      slots: {
        default: () =>
          h(NCard, null, {
            default: () => 'test'
          })
      }
    })
    expect(wrapper.find('.n-layout').attributes('style')).toMatchSnapshot()
  })

  it('should work with `bordered` prop', async () => {
    const wrapper = mount(NLayout, {
      props: {
        'has-sider': true
      },
      slots: {
        default: () => [
          h(
            NLayoutHeader,
            { bordered: true },
            { default: () => 'test-header' }
          ),
          h(NLayoutSider, { bordered: true }, { default: () => 'test-sider' }),
          h(NLayoutFooter, { bordered: true }, { default: () => 'test-footer' })
        ]
      }
    })
    expect(wrapper.find('.n-layout-header').classes()).toContain(
      'n-layout-header--bordered'
    )
    expect(wrapper.find('.n-layout-sider').classes()).toContain(
      'n-layout-sider--bordered'
    )
    expect(wrapper.find('.n-layout-footer').classes()).toContain(
      'n-layout-footer--bordered'
    )
  })

  it('should work with `absolute` prop', async () => {
    const wrapper = mount(NLayout, {
      props: {
        position: 'absolute'
      },
      slots: {
        default: () => [
          h(NLayoutHeader, null, { default: () => 'test-header' })
        ]
      }
    })
    expect(wrapper.find('.n-layout').classes()).toContain(
      'n-layout--absolute-positioned'
    )
  })
})
