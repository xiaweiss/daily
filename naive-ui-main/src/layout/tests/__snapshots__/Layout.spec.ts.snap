// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`n-layout should work with \`embedded\` prop 1`] = `"--bezier: cubic-bezier(.4, 0, .2, 1); --color: rgb(250, 250, 252); --text-color: rgb(51, 54, 57);"`;
